### PR TITLE
Issue 31 controller path to swagger

### PR DESCRIPTION
**Pull request by berenda-andrey**
Closes https://github.com/mofr/winter/issues/31

### CODE REVIEW

File: winter/controller.py [hunk 1]
```diff
@@ -29,10 +33,21 @@ def __repr__(self):
 
 
 class ControllerMethod:
-    def __init__(self, func, url_path: str, http_method: str):
+
+    def __init__(
+            self,
+            controller_cls: type,
```
Comment (berenda-andrey) [inline winter/controller.py]: Add this argument

File: winter/controller.py [hunk 2]
```diff
@@ -71,26 +93,33 @@ def parameter(self) -> inspect.Parameter:
         return self.method.signature.parameters[self.name]
 
 
-def _register_controller(controller_class):
+def _register_controller(controller_class: type) -> None:
```
Comment (pristupa) [inline winter/controller.py]: I like the idea that `Type` is for typing, `type` is for code - `type()`

File: winter/schema/path_parameters_inspector.py [hunk 3]
```diff
@@ -16,14 +16,14 @@ def inspect_parameters(self, controller_method: ControllerMethod) -> List[openap
 
         parameters = []
 
-        for path_variable_name in uritemplate.variables(controller_method.url_path):
+        for path_variable_name in uritemplate.variables(controller_method.full_url_path):
```
Comment (berenda-andrey) [inline winter/schema/path_parameters_inspector.py]: This solve issue-31

File: winter/schema/generation.py [hunk 4]
```diff
@@ -54,7 +54,7 @@ def get_argument_type_info(argument: ControllerMethodArgument) -> dict:
     try:
         type_info = inspect_type(argument.type_)
     except InspectorNotFound:
-        return {'type': openapi.TYPE_STRING}
+        return None
```
Comment (berenda-andrey) [inline winter/schema/generation.py]: It was my fault

File: winter/http_method.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import enum
+
+
+class HttpMethod(enum.Enum):
```
Comment (pristupa) [inline winter/http_method.py]: HTTPMethod
PEP8 :(

File: winter/routing.py [hunk 6]
```diff
@@ -1,46 +1,50 @@
...
+    return route(url_path, HttpMethod.PUT)
 
 
 def register_route(func, url_path, http_method):
     assert func not in _routes, f'{func} is already mapped to a route'
     _routes[func] = Route(url_path, http_method)
 
 
-def get_function_route(func) -> Optional[Route]:
+def get_route(func: typing.Union[type, typing.Callable]) -> typing.Optional[Route]:
```
Comment (berenda-andrey) [inline winter/routing.py]: sometimes it is not function, sometimes controller

File: winter/response_status.py [hunk 7]
```diff
@@ -1,25 +1,26 @@
...
+from .http_method import HttpMethod
+
+_default_statuses: typing.Dict[typing.Callable, int] = {}
 _default_http_method_statuses = {
-    'get': 200,
-    'post': 200,
-    'patch': 200,
-    'delete': 204,
+    HttpMethod.GET: 200,
+    HttpMethod.POST: 200,
```
Comment (pristupa) [inline winter/response_status.py]: 201?

File: winter/controller.py [hunk 8]
```diff
@@ -29,10 +33,21 @@ def __repr__(self):
...
 class ControllerMethod:
-    def __init__(self, func, url_path: str, http_method: str):
+
+    def __init__(
+            self,
+            controller_cls: type,
+            func,
+            url_path: str,
+            http_method: HttpMethod,
+            default_response_status: typing.Optional[int] = 200
```
Comment (berenda-andrey) [inline winter/controller.py]: Also add this argument

File: winter/http_method.py (hunk 5 above)
Comment (berenda-andrey) [reply winter/http_method.py]: Renamed

File: winter/response_status.py (hunk 7 above)
Comment (berenda-andrey) [reply winter/response_status.py]: may be, but we need discuss it

File: winter/controller.py (hunk 2 above)
Comment (berenda-andrey) [reply winter/controller.py]: Dome

File: winter/schema/path_parameters_inspector.py (hunk 3 above)
Comment (mofr) [reply winter/schema/path_parameters_inspector.py]: Type it like this to automatically close the issue on PR merge: Closes #31

**Pull request closed without merging**